cmd/podman/root.go

```diff
@@ -185,15 +191,17 @@ func readRemoteCliFlags(cmd *cobra.Command, podmanConfig *entities.PodmanConfig)
 // 2. Env variables (CONTAINER_HOST and CONTAINER_CONNECTION);
 // 3. ActiveService from containers.conf;
 // 4. RemoteURI;
-func setupRemoteConnection(podmanConfig *entities.PodmanConfig) error {
+// Returns the name of the default connection if any.
+func setupRemoteConnection(podmanConfig *entities.PodmanConfig) string {
 	conf := podmanConfig.ContainersConfDefaultsRO
 	connEnv, hostEnv, sshkeyEnv := os.Getenv("CONTAINER_CONNECTION"), os.Getenv("CONTAINER_HOST"), os.Getenv("CONTAINER_SSHKEY")
 
 	switch {
 	case connEnv != "":
 		con, err := conf.GetConnection(connEnv, false)
 		if err != nil {
-			return err
+			podmanConfig.ConnectionError = err
+			return connEnv
```

Comment:
Should this be `return ""`? It probably doesn't matter though.

Reply:
No, if you run `CONTAINER_CONNECTION=abc podman --help` it should list the connection name as default regardless of it is valid or not I would say.

Although now that I look at it I missed the `return con.Name` after line 208 to have the name in the success case.

Reply:
Hm, I thought you were arguing against having the environment variables be used as what appears in the defaults in `--help`?

But I don't have a strong opinion, up to you.

Reply:
I am not the biggest fan but this is what we do with CONTAINER_HOST/CONTAINER_SSHKEY already so I prefer to just match that for consistency. And anyhow the current default listed was just incorrect depending on the configuration so I thought might as well fix that while I touch the code there.